numpy/random/tests/test_random.py

```diff
@@ -505,6 +505,32 @@ def test_shuffle_masked(self):
             assert_equal(
                 sorted(b.data[~b.mask]), sorted(b_orig.data[~b_orig.mask]))
 
+    def test_permutation(self):
+        np.random.seed(self.seed)
+        alist = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
+        actual = np.random.permutation(alist)
+        desired = [0, 1, 9, 6, 2, 4, 5, 8, 7, 3]
+        assert_array_equal(actual, desired)
+
+        np.random.seed(self.seed)
+        arr_2d = np.atleast_2d([1, 2, 3, 4, 5, 6, 7, 8, 9, 0]).T
+        actual = np.random.permutation(arr_2d)
+        assert_array_equal(actual, np.atleast_2d(desired).T)
+
+        # test permutation on a different axis
+        a = np.array([[1, 2, 3, 4, 5],
+                      [2, 3, 4, 5, 6],
+                      [3, 4, 5, 6, 7],
+                      [4, 5, 6, 7, 8]])
+
+        c = np.array([[1, 2, 5, 4, 3],
+                      [2, 3, 6, 5, 4],
+                      [3, 4, 7, 6, 5],
+                      [4, 5, 8, 7, 6]])
+        np.random.seed(self.seed)
+        actual = np.random.permutation(a, axis=1)
```

Comment:
Also do the same test with `axis=-4` to make sure you get the same thing as you do with `axis=1`.

Reply:
You are correct, we should also test negative axis. The tested array is a 2d array, maybe `axis=-1`, since it should be `-ndim <= axis < ndim` for `normalize_axis_index`?

Reply:
You're right. I conflated axis and column.

Reply:
Updated! :)